// webpack-bundle-analyzer/lib/tree/ContentFolder.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _BaseFolder = _interopRequireDefault(require("./BaseFolder"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
class ContentFolder extends _BaseFolder.default {
  constructor(name, ownerModule, parent) {
    super(name, parent);
    this.ownerModule = ownerModule;
  }
  get parsedSize() {
    return this.getSize('parsedSize');
  }
  get gzipSize() {
    return this.getSize('gzipSize');
  }
  get brotliSize() {
    return this.getSize('brotliSize');
  }
  getSize(sizeType) {
    const ownerModuleSize = this.ownerModule[sizeType];
    if (ownerModuleSize !== undefined) {
      return Math.floor(this.size / this.ownerModule.size * ownerModuleSize);
    }
  }
  toChartData() {
    return {
      ...super.toChartData(),
      parsedSize: this.parsedSize,
      gzipSize: this.gzipSize,
      brotliSize: this.brotliSize,
      inaccurateSizes: true
    };
  }
}
exports.default = ContentFolder;
;
